src/components/Datatable/Pagination/Pagination.tsx

```diff
@@ -0,0 +1,206 @@
+/* eslint-disable react/boolean-prop-naming */
+import React, { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+import { getColor, pxToRem } from '../../../utils/helpers';
+import { FlexContainer } from '../../FlexContainer';
+import { SSCIconNames } from '../../../theme/icons/icons.enums';
+import { Spinner } from '../../Spinner';
+import { Label } from '../../forms/Label';
+import { PaginationProps } from './Pagination.types';
+import PageButton from './PageButton';
+import NavButton from './NavButton';
+import { StyledNumber } from '../../Filters/components/Number/Number';
+
+const MAX_PAGE_BUTTONS = 8;
+
+const PaginationWrapper = styled(FlexContainer)`
+  background-color: ${getColor('graphite3H')};
+  padding: ${pxToRem(16)};
+`;
+
+const Ellipsis = styled.div`
+  display: inline-block;
+  width: ${pxToRem(48)};
+  height: ${pxToRem(32)};
+  padding: ${pxToRem(8, 16)};
+  margin: ${pxToRem(0, 4)};
+  color: ${getColor('graphite2B')};
+  background-color: ${getColor('transparent')};
+`;
+
+const PageInputLabel = styled(Label)`
+  margin: ${pxToRem(6, 16)};
+`;
+const PageInput = styled(StyledNumber)`
+  width: ${pxToRem(72)};
+  height: ${pxToRem(32)};
+  padding: ${pxToRem(8, 16)};
+`;
+
+const Pagination: React.FC<PaginationProps> = ({
+  gotoPage,
```

Comment:
Could you add `on` prefix to fn props? `onGoToPage, onPreviousPage, onNextPage`

Reply:
I would prefer to keep the names in this case because they come directly from react-table.

Reply:
Pagination is our Component and to be consistent it should follow the same practices as other components so handlers should be prefixed with `on*` we can move the pagination pattern outside of Datatable component in the future and we would need to refactor this

Reply:
Our components, our rules 🙂